### PR TITLE
BGP neighbor puppet support for XR

### PR DESCRIPTION
**Pull request by chburket**

### CODE REVIEW

File: lib/cisco_node_utils/bgp_neighbor.rb [hunk 1]
```diff
@@ -406,7 +406,8 @@ def transport_passive_mode=(val)
           config_set('bgp_neighbor', 'transport_passive_mode', @set_args)
```
Comment (RichWellum) [inline lib/cisco_node_utils/bgp_neighbor.rb:406]: :+1:

File: lib/cisco_node_utils/bgp_neighbor.rb [hunk 2]
```diff
@@ -431,7 +432,7 @@ def default_transport_passive_mode
 
     def transport_passive_only=(val)
       # TODO: add "deprecated" warning?
-      self.transport_passive_mode = (val ? :passive_only : :both)
+      self.transport_passive_mode = (val ? :passive_only : :none)
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/bgp_neighbor.rb:435]: The getters wrap this in a `:nexus` check - is the setter safe for both platforms?

File: lib/cisco_node_utils/bgp_neighbor.rb [hunk 3]
```diff
@@ -469,6 +470,8 @@ def mode_symbol_to_cli(symbol)
         'passive-only'
       when :both
         'both'
+      when :none
+        ''
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/bgp_neighbor.rb:474]: I don't know the background behind this commit but on the surface I'm wondering is `:both` is still relevant after the commit? The getters and setters can't use it so can it be removed altogether?

File: tests/test_bgp_neighbor.rb [hunk 4]
```diff
@@ -483,10 +483,11 @@ def test_transport_passive_mode
       neighbor = create_neighbor(vrf)
       check = []
       if platform == :ios_xr
-        check = [:active_only, :passive_only, :both,
+        check = [:active_only, :passive_only, :both, :none,
```
Comment (chrisvanheuveln) [inline tests/test_bgp_neighbor.rb:486]: Remove `:both` ?

File: lib/cisco_node_utils/bgp_neighbor.rb (hunk 2 above)
Comment (chburket) [inline lib/cisco_node_utils/bgp_neighbor.rb:435]: The transport_passive_mode setter also has a :nexus check.  Maybe you are looking at the transport_passive_only setter?

File: lib/cisco_node_utils/bgp_neighbor.rb (hunk 3 above)
Comment (chburket) [inline lib/cisco_node_utils/bgp_neighbor.rb:474]: :both is still relevant for XR in the transport_passive_mode methods.

File: tests/test_bgp_neighbor.rb (hunk 4 above)
Comment (chburket) [inline tests/test_bgp_neighbor.rb:486]: :both is still valid for XR.  The default has been changed to :none (which will clear the setting).

File: lib/cisco_node_utils/bgp_neighbor.rb (hunk 2 above)
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/bgp_neighbor.rb:435]: yep, the method names blurred for me. nm.

File: lib/cisco_node_utils/bgp_neighbor.rb (hunk 3 above)
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/bgp_neighbor.rb:474]: ack

**Pull request merged**